src/other/Persist.js

```diff
@@ -280,6 +280,7 @@
             if (typeof state === "string") {
                 state = JSON.parse(state);
             }
+            state = Utility.retrofit_114_serialization(state,new Platform().version());
```

Comment:
The retrofit code belongs in this source file - and if so it doesn't need to use any Platform().version test at all (as we know it has to be > 1.14.

IOW If the viz framework in use is >= v1.20.0 and we are passed a v1.14 persist string, then we do the upgrade on the fly.

The responsibility of initializing the correct viz framework version resides externally.

Reply:
The `Platform().version` test gives us the target version for the upgrade script. The only alternative that I can imagine is a hard-coded default target? (probably 1.18.0)